examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1245 @@
+      double difference_norm_square = 0;
+      for (unsigned int point=0; point<q_points.size(); ++point)
+        {
+          const double diff= (g[point]-sol_u[point]);
+          difference_norm_square += diff*diff*JxW[point];
+        }
+        copy_data.value +=std::sqrt(penalty*difference_norm_square);
+    };
+
+    auto face_worker = [&] (const Iterator &cell, 
+                            const unsigned int &f, 
+                            const unsigned int &sf,
+                            const Iterator &ncell, 
+                            const unsigned int &nf,
+                            const unsigned int &nsf,
+                            ScratchData1<dim> &scratch_data,
+                            CopyData &copy_data)
+    {
+      FEInterfaceValues<dim> &fe_iv = scratch_data.fe_interface_values;
+      fe_iv.reinit(cell, f, sf, ncell, nf, nsf);  
+      
+      copy_data.face_data.emplace_back();
+      CopyDataFace &copy_data_face = copy_data.face_data.back();
+      
+      copy_data_face.cell_indices[0] = cell->active_cell_index();
+      copy_data_face.cell_indices[1] = ncell->active_cell_index();
+
+      const std::vector<double> &JxW = fe_iv.get_JxW_values ();
+      const std::vector<Tensor<1,dim> > &normals = fe_iv.get_normal_vectors ();
+      
+      const auto &q_points = fe_iv.get_quadrature_points();
+      const unsigned int n_q_points = q_points.size();
+
+      std::vector<double> nu (n_q_points);
+      viscosity_function.value_list (q_points, nu);
+      
+      std::vector<Tensor<1, dim>> grad_u[2];
+      std::vector<double> sol_u[2];
+
+      const double h = cell->face(f)->measure();
+      for(unsigned int i = 0; i < 2; ++i)
+      {
+        grad_u[i].resize(n_q_points);
+        sol_u[i].resize(n_q_points);
+        fe_iv.get_fe_face_values(i).get_function_values(
+        solution, sol_u[i]);
+        fe_iv.get_fe_face_values(i).get_function_gradients(
+        solution, grad_u[i]);
+      }
+
+      const double degree = std::max(1.0, static_cast<double>(fe_iv.get_fe().degree));
+      const double extent1 = cell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[f])
+          * (cell->has_children() ? 2.0 : 1.0);
+      const double extent2 = ncell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[nf])
+          * (ncell->has_children() ? 2.0 : 1.0);
+      const double penalty =4.0 * degree * (degree+1.0) * 0.5 * (1.0/extent1 + 1.0/extent2);
+
+      double flux_jump_square=0;
+      double u_jump_square =0;
+      for (unsigned int point=0; point<n_q_points; ++point)
+        {
+          const double u_jump = sol_u[0][point]-sol_u[1][point];
+          u_jump_square += u_jump*u_jump*JxW[point];
+
+          const double flux_jump = (grad_u[0][point]-grad_u[1][point])*normals[point];
+          flux_jump_square += nu[point]*flux_jump*flux_jump*JxW[point];
+        }
+        copy_data_face.values[0] = 0.5*std::sqrt(h*flux_jump_square + penalty*u_jump_square);
```

Comment:
we should look into using the MeshWorker::CopyData but this won't work directly as it only contains matrices and vectors, hm. We need to talk about that.

Reply:
Maybe leave it as is for now but leave a TODO